core/src/com/fourinachamber/fortyfive/screen/general/CustomActors.kt

```diff
@@ -533,6 +532,282 @@ open class CustomFlexBox(
     }
 }
 
+class CustomScrollableFlexBox(
+    private val screen: OnjScreen,
+    private val isScrollDirectionVertical: Boolean,
+    private val scrollDistance: Float,
+    private val isBackgroundStretched: Boolean,
+    private val scrollbarBackgroundName: String?,
+    private val scrollbarName: String?,
+    private val scrollbarSide: String?,
+) : CustomFlexBox(screen) {
+
+    private val scrollListener = object : InputListener() {
+        override fun enter(event: InputEvent?, x: Float, y: Float, pointer: Int, fromActor: Actor?) {
+            stage.scrollFocus = this@CustomScrollableFlexBox
+        }
+
+        override fun exit(event: InputEvent?, x: Float, y: Float, pointer: Int, toActor: Actor?) {
+            if (x > width || x < 0 || y > height || y < 0)
+                stage.scrollFocus = null
+        }
+
+        override fun scrolled(event: InputEvent?, x: Float, y: Float, amountX: Float, amountY: Float): Boolean {
+            this@CustomScrollableFlexBox.scroll(amountY)
+            return super.scrolled(event, x, y, amountX, amountY)
+        }
+    }
+
+    private var needsScrollbar: Boolean = true
+
+    private val dragListener = object : DragListener() {
+
+        private var startPos by Delegates.notNull<Float>()
+
+        override fun dragStart(event: InputEvent?, x: Float, y: Float, pointer: Int) {
+            val parentPos = this@CustomScrollableFlexBox.localToStageCoordinates(Vector2(0, 0))
+            startPos = if (isScrollDirectionVertical) {
+                val relX = (if (scrollbarSide == "left") 0F else width - scrollbarHandle.width)
+                val relY = scrollbarHandle.y - parentPos.y
+                if (touchDownX < relX + scrollbarHandle.width && touchDownX > relX
+                    && touchDownY < relY + scrollbarHandle.height && touchDownY > relY
+                ) scrollbarHandle.y
+                else Float.NaN
+            } else {
+                val relX = scrollbarHandle.x - parentPos.x
+                val relY = (if (scrollbarSide == "top") height - scrollbarHandle.height else 0F)
+                if (touchDownX < relX + scrollbarHandle.width && touchDownX > relX
+                    && touchDownY < relY + scrollbarHandle.height && touchDownY > relY
+                ) scrollbarHandle.x
+                else Float.NaN
```

Comment:
Also, I typically try to avoid using NaN where it is possible, because of inituitive behaviour. In this case, it could be better to use something like null instead